datalad/support/annexrepo.py

```diff
@@ -594,40 +594,48 @@ def check_repository_versions(cls):
 
     @staticmethod
     def get_size_from_key(key):
-        """A little helper to obtain size encoded in a key"""
+        """A little helper to obtain size encoded in a key
+
+        Returns
+        -------
+        int or None
+          size of the file or None if either no size is encoded in the key or
+          key was None itself
+
+        Raises
+        ------
+        ValueError
+          if key is considered invalid (at least its size-related part)
+        """
         if not key:
             return None
 
-        # TODO: this method can be more compact. we don't need particularly
-        #       elaborated error distinction
-
         # see: https://git-annex.branchable.com/internals/key_format/
         key_parts = key.split('--')
         key_fields = key_parts[0].split('-')
-
-        s = S = C = None
-
-        for field in key_fields[1:]:  # first one has to be backend -> ignore
-            if field.startswith('s'):
-                # size of the annexed file content:
-                s = int(field[1:]) if field[1:].isdigit() else None
-            elif field.startswith('S'):
-                # we have a chunk and that's the chunksize:
-                S = int(field[1:]) if field[1:].isdigit() else None
-            elif field.startswith('C'):
-                # we have a chunk, this is it's number:
-                C = int(field[1:]) if field[1:].isdigit() else None
-
-        if s is None:
-            return None
-        elif S is None and C is None:
-            return s
+        parsed = {field[0]: int(field[1:]) if field[1:].isdigit() else None
+                  for field in key_fields[1:]
+                  if field[0] in "sSC"}
+
+        # don't lookup the dict for the same things several times;
+        # Is there a faster (and more compact) way of doing this? Note, that
+        # locals() can't be updated.
+        s = parsed['s'] if 's' in parsed.keys() else None
```

Comment:
`parsed.get('s')`?

---

I know this was in response to a review comment from Yarik, but I must admit that I don't find the newer version to be an improvement in terms of readability.  Perhaps a compromise between this and the previous form would be an internal function that encapsulates `int(field[1:]) if field[1:].isdigit() else None` while keeping the expanded `ifelif` branches.

Anyway, it's a minor thing either way, and I'm fine with how it is.  (I'm just glad you didn't use `locals()` :] )

Reply:
> `parsed.get('s')`

WTH? Why didn't I think of it? :-D

Reply:
Changed to `get`, but otherwise will leave as is. Kind of agree with the readability aspect, but it's not too bad and TBH I don't really like additional function calls for that.